sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -366,7 +366,7 @@ ssize_t coap_opt_get_string(coap_pkt_t *pkt, uint16_t optnum,
     return (int)(max_len - left);
 }
 
-int coap_get_blockopt(coap_pkt_t *pkt, uint16_t option, uint32_t *blknum, unsigned *szx)
+int coap_get_blockopt(coap_pkt_t *pkt, uint16_t option, uint32_t *blknum, uint8_t *szx)
```

Comment:
```suggestion
int coap_get_blockopt(coap_pkt_t *pkt, uint16_t option, coap_block1_t * block)
```

Reply:
That would be a breaking API change

Reply:
as changing unsigned * to uint8 * is (for unitized usigned szx it may not behave nicely).

Reply:
Ah but we don't always want a `coap_block1_t`.
The users of `coap_get_blockopt()` are converted now though, so this shouldn't be an issue.

Reply:
since we already have an api change we can as well change to `coap_block_t *` 

-- `typedef coap_block1_t coap_block_t` i don't know why i has this typename (there is no differenc in the actual information of a coap_block1 an a coap_block2 (as indicated by using the same function for both)

Reply:
But not all users of `coap_get_blockopt()` want the struct - we'd needlessly have to put it on the stack then just to extract those values.

Reply:
putting two values on the stack or putting a struct (containing said values) on the stack is about the same code.
But the call with just one pointer (just to the struct) is cheaper that with two (need less registers or stack for the call -> less to free before the call). even if the next function needs both values separated (the slicer init case) it is cheaper to use a struct.

There is only the case of an external structure that you may want to directly write the values into but in that case
you would assume the user has to change his structure which would make this upgrade this PR to even more api changing (user structures -> user code).

Reply:
https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGIMxqkrgAyeAyYAHI%2BAEaYxP6kAA6oCoRODB7evv6ByamOAqHhUSyx8Wa2mPYFDEIETMQEmT5%2BAZXV6XUNBEWRMXEJCvWNzdltQ929JWX%2BAJS2qF7EyOwc5gDMYcjeWADU5mZD%2BIIAdAgHJhoAghtbO5j7ZocExGHAZxfXlzdm61g04V2bgA8ldlAB9CzBYFuADSAHUAJJCbDgiJyACy4OBADEcbtdhBJLNvht/mEHiCwZDoXCkSjwRjgQAlVG4/EQdYkr6/cmAqkQqEwhHI1FCABaAA1GVchLDCRpVBoAOzcn5/TAAymgwW0kUMiXSjFXSUEiCq75eMIEdZmcEEXZUBgQK2CABskntuwAVIkANYEUi7V0ELhur2oRI1EnrKzXAkE77W3ZoJiJcHATAEcHRWioZB%2ByMEF3Wj1e30BoMhsMRqPpKvW23l3N%2Bhg%2BBuCAAc5YUAC9VGqTMqLAnvgng427Q6W0X9usACKO53%2BwNF9IxuNXcdjhPelttlhzxczqNz7AmdbYIE6mnC%2Bmo9FY9kXzfj7191RH3Ynh3mN3X6khTpUVwUNGU5RfUl4wTYgsyWBhCR/R5/0VTtZjnfEuF2EBdg0SCvmVecoKuAgAE9Ekwf5diGYgvAcfZhwnQQm2nWhWx8F8mIIbsHQ/SDCO/PMC3tfCbmuZNU3TTNs1zfNCyjMwS3dT0HQrQMuJrB01wEUhZOE1TokHRjE2ggkQxYwT5N/BclwgFdSG0hgNx3EytwTaIAFoL2wfcfC/JDvO8gDdTvEDH2xPFRPHTzvI/fy2NnP9gtvYCDSlcDYVElzdlggh4MQhLTyS1D0IvTDsNwrLCOIgA3VA8HQb9UGicEwkIJSbSnSyDw7bivQ/ZzrhDMtVN2FdONq%2BrGq8JRiC4CA6oaklh2y8yut8lhOJDHjqP7KKE0kjMsxzISrLsytdi4IM/w23ZSD/OLBrcgloma1qGHajagwGqqiK%2Ba5Fum2bFMB5bX3c06vRbPB9oJQ7pJOuSi0U%2BzLuusw3Whp7oretri2hk4voJn7Y1Jarrg4eZaE4ABWXg/A4LQHM4NxrGsajFmWB4Nh4UgCE0Sn5j9EAacCamOEkemBeZjheAUEBAn5xnKdIOBYCQNAWESOg4nIShNe1%2Bh4gaFguC4NoaFoAg4nliBoml6IwgaUjOF5zW2EEYEGFoF3ldILAWEMYBxD9/BYIcPAasweW/cwVRMGQLxrdd3hrSqaXaDwaJiGdjwsGll48BYFP5ioAxgAUAA1PBMAAd2BCiGd5/hBBEMR2CkGRBEUFR1D93QroMIwUDZyx9Cz%2BXIHmRyY484EzF2DynVQDywkz8I5aqBOahcBh3E8Fo9BCClpgGK68jSARRj8c%2BUkvhgpn6eIrrsbfOmGJoD%2ByF%2Bt4jgQukaI/UoZ9bAf2vnoCYgCT5PwkPMBQnMVh6BeJgVYPAqa0yln7VAnArjMgxLsYAyBkCXS4CcBeEB15eAHISVmlhrBBlwIQEgjx1hXSBKgLWOtiAsK4LMXgSstCzCFiLMWnBJakAZkzbBstbAKz5gLIR%2BhOBmEwVIzg/CFHzCjsQVIzhJBAA

Reply:
But the struct contains other members too

Reply:
even with the two extra members in the block_t the struct variant is still better just not as much (takes 8 extra stack bytes a little earlier (the other variant still needs them since it needs registers (with are in use by the pointers and they go on the stack).

Reply:
The struct ends up 4 bytes smaller: https://godbolt.org/z/P7enh1Y5c

Reply:
my second godblot example shows a alignment thing going on  the structure decreased in size  and if another variable will go on stack that extra space is avail see https://godbolt.org/z/Trab6eoY9 
While `t_a` and `t_b` use the same amount of stack `T_a` and `T_b` don't